x-pack/plugins/apm/public/components/app/trace_overview/index.tsx

```diff
@@ -64,6 +64,7 @@ export function TraceOverview({ children }: { children: React.ReactElement }) {
       transactionId: '',
       detailTab: TransactionTab.timeline,
       showCriticalPath: false,
+      flyoutDetailTab: '',
```

Comment:
I made the `flyoutDetailTab` not required for the /trace/explorer/waterfall path, as it is for the others (dependencies, transactions view etc) in this [commit](https://github.com/elastic/kibana/pull/148309/commits/ed9f4a278fbb9f9735e01e332aea0a885c9f8913)

> Instead of removing unwanted params we might want to not apply ...query and instead only pass params explicitly.

We might also want to set as not required `traceId`, `transactionId` and `waterfallItemId`

Reply:
> We might also want to set as not required `traceId`, `transactionId` and `waterfallItemId`

I recall a comment from @dgieselaar that he prefers having query params required (even if they are not) to ensure that they are passed - it's easy to forget to pass optional params. 

That being said, I like how it is now. It's the smaller change. So let's keep it like that and change later if we decide it should be required.